app/models/concerns/profile_searching.rb

```diff
@@ -148,6 +149,33 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
         conditions: "profiles.id IN (#{any_child_profile.to_sql})"
       }
     end
+
+    def filter_by_policy_and_profile_names(_filter, operator, value)
+      values = preprocess_parameters(operator, value)
+      cond = sanitize_sql_for_conditions(["policies.name #{operator} (?) OR profiles.name #{operator} (?)",
+                                          values, values])
```

Comment:
Using raw SQL is always the worst possible option, you can write this nicer by using the [AR Query Interface](https://guides.rubyonrails.org/active_record_querying.html#or-conditions) at least partially. Or if that's not an option, there's always Arel. Something like this should work:

```
left_outer_joins(:policy).where("policies.name #{operator} ?", values).or(Profiles.where('profiles.name #{operator} ?', values))
```